Library/OcPciIoLib/OcPciIoU.c

```diff
@@ -0,0 +1,446 @@
+/** @file
+   EFI PCI IO protocol functions implementation for PCI Bus module.
+
+   Copyright (c) 2006 - 2019, Intel Corporation. All rights reserved.<BR>
+   Copyright (c) 2023, xCuri0. All rights reserved.<BR>
+   SPDX-License-Identifier: BSD-2-Clause-Patent
+
+ **/
+
+#include "OcPciIoU.h"
+
+EFI_CPU_IO2_PROTOCOL  *mCpuIo;
+
+UINT8  mInStride[] = {
+  1,      // EfiCpuIoWidthUint8
+  2,      // EfiCpuIoWidthUint16
+  4,      // EfiCpuIoWidthUint32
+  8,      // EfiCpuIoWidthUint64
+  0,      // EfiCpuIoWidthFifoUint8
+  0,      // EfiCpuIoWidthFifoUint16
+  0,      // EfiCpuIoWidthFifoUint32
+  0,      // EfiCpuIoWidthFifoUint64
+  1,      // EfiCpuIoWidthFillUint8
+  2,      // EfiCpuIoWidthFillUint16
+  4,      // EfiCpuIoWidthFillUint32
+  8       // EfiCpuIoWidthFillUint64
+};
+
+//
+// Lookup table for increment values based on transfer widths
+//
+UINT8  mOutStride[] = {
+  1,      // EfiCpuIoWidthUint8
+  2,      // EfiCpuIoWidthUint16
+  4,      // EfiCpuIoWidthUint32
+  8,      // EfiCpuIoWidthUint64
+  1,      // EfiCpuIoWidthFifoUint8
+  2,      // EfiCpuIoWidthFifoUint16
+  4,      // EfiCpuIoWidthFifoUint32
+  8,      // EfiCpuIoWidthFifoUint64
+  0,      // EfiCpuIoWidthFillUint8
+  0,      // EfiCpuIoWidthFillUint16
+  0,      // EfiCpuIoWidthFillUint32
+  0       // EfiCpuIoWidthFillUint64
+};
+
+EFI_CPU_IO2_PROTOCOL *
+InitializeCpuIo2 (
+  )
+{
+  mCpuIo = NULL;
```

Comment:
- Please cache the value. If `mCpuIo` is not null, then return it.
- Check the return value of `gBS->LocateProtocol` and return NULL explicitly on failure.